effekt/shared/src/main/scala/effekt/lifted/Tree.scala

```diff
@@ -119,12 +119,25 @@ enum Stmt extends Tree  {
 
   // Local Control Flow
   case If(cond: Expr, thn: Stmt, els: Stmt)
-  case Match(scrutinee: Expr, clauses: List[(Symbol, BlockLit)], default: Option[Stmt])
+  case Match(scrutinee: Expr, clauses: List[(Id, BlockLit)], default: Option[Stmt])
 
   // Effects
-  case State(id: Id, init: Expr, region: Symbol, ev: Evidence, body: Stmt)
-  case Try(body: Block, handler: List[Implementation])
+
+  // allocates into a (type-monomorphic?) region.
+  // e.g. var x in r = init; body
   case Region(body: Block)
+  case Alloc(id: Id, init: Expr, region: Id, ev: Evidence, body: Stmt)
+
+  // creates a fresh state handler to model local (backtrackable) state.
+  // e.g. state(init) { (ev){x: Ref} => ... }
+  case Var(init: Expr, body: Block.BlockLit)
+  case Get(id: Id, ev: Evidence, annotatedTpe: ValueType)
+  case Put(id: Id, ev: Evidence, value: Expr)
```

Comment:
The fact that `Get` and `Put` have an `id : Id` confused me a lot. These should be `Expr`.

Reply:
That's right -- for regions it could be general expressions. For local variables it is always just names.

Reply:
Long story short: References are not (yet) first class citizens.